x-pack/plugins/infra/public/pages/metrics/hosts/components/common/popover.tsx

```diff
@@ -5,17 +5,15 @@
  * 2.0.
```

Comment:
popover components are duplicates because asset details will be moved to a separate plugin/package at some point. When that happens, we can think about what can be done to reuse this component.

Reply:
Why would we want to reuse this component? AFAICT this is just a wrapper component to use the EuiPopover. IMHO we should reuse the EuiPopover and not a wrapper we created. This can easily start growing to an unmanageable state as custom logic is added for different places. Making it hard to test and refactor in the future.

Reply:
It's not mandatory to reuse a single component on both pages. Both popover components share pretty similar behavior and encapsulate the default props we want the popover to have. It makes maintainability easier if you consider we have 10+ popovers across these pages.

Reply:
>It makes maintainability easier if you consider we have 10+ popovers across these pages.

Until it doesn't.

Reply:
hm. Could you elaborate? I can't see this component growing as you've mentioned in your comment above.

Reply:
My point is that the `EuiPopover` is already an abstraction. It encapsulates the hard job of showing a popover. This component is another layer of abstraction that only handles the open/close state AFAICT.  Soon we'll probably have a component that uses this one and needs to load a different icon or render a different size or position or any other property, then we start adding properties to this component for that specific case.

What I think is we should share components that make sense to be shared and this one IMHO is not.

Reply:
I see your point. It's really hard to predict whether such a thing will happen or not, though. We can only assume what we have now, which is this simple usage. If this turns out to be a real thing, we can think of a different approach, refactor it, and do something different.

Regarding maintainability, I would have had to make this change in multiple files instead of a couple of files. Needless to say the open/closed state and default props would also have to be repeated everywhere that's using a popover here.

As for sharing components, it's not in the scope of this PR to create a single wrapper component for the `EuiPopover` to be shared across infra, and cover N different scenarios, and this might not even be needed tbh.